### ISSUE TITLE
Enums incorrectly exported in Nativescript 8

### ISSUE THREAD

**Issue by PeterStaev** [NativeScript/NativeScript#9298]
**Environment**
Provide version numbers for the following components (information can be retrieved by running `tns info` in your project folder or by inspecting the `package.json` of the project):
 - CLI: 8.0.1
 - Cross-platform modules: 8.0
 - Android Runtime: 8.0
 - iOS Runtime: 8.0
 - XCode Version: 12.4
 - Plugin(s):

**Describe the bug**
Trying to update a project to NS8, and seems the `Enums` object is messed up. I see that it is deprecated and `CoreTypes` should be used, which works nicely when i changed in my app. But plugins that were using it are no longer compatible. So it is not deprecated but more or less removed and is REQUIRED to be changed in all plugins. Couple of examples:
```
WARNING in ./node_modules/@nativescript/firebase/admob/admob.ios.js 277:45-68
export 'Enums' (imported as 'Enums') was not found in '@nativescript/core' (possible exports: AbsoluteLayout, AccessibilityEvents, AccessibilityLiveRegion, AccessibilityRole, AccessibilityState, AccessibilityTrait, ActionBar, ActionItem, ActionItems, ActivityIndicator, AndroidApplication, Animation, AnimationCurve, Application, ApplicationSettings, Background, Binding, Builder, Button, CSSHelper, CSSType, CSSUtils, ChangeType, CoercibleProperty, Color, Connectivity, ContainerView, ContentView, ControlStateChangeListener, CoreTypes, CssAnimationParser, CssAnimationProperty, CssProperty, CustomLayoutView, DatePicker, Device, DialogStrings, Dialogs, DockLayout, EditableTextBase, File, FileSystemEntity, FlexboxLayout, Folder, Font, FontScaleCategory, FormattedString, Frame, GestureStateTypes, GestureTypes, GesturesObserver, GridLayout, GridUnitType, HtmlView, Http, IOSHelper, Image, ImageAsset, ImageCache, ImageSource, InheritedCssProperty, InheritedProperty, ItemSpec, KeyframeAnimation, KeyframeAnimationInfo, KeyframeDeclaration, KeyframeInfo, Label, LayoutBase, Length, ListPicker, ListView, ModuleNameResolver, NavigationButton, Observable, ObservableArray, Page, PageBase, ParserEvent, ParserEventType, PercentLength, Placeholder, Progress, Property, ProxyViewContainer, PseudoClassHandler, Repeater, RootLayout, Screen, ScrollView, SearchBar, SegmentedBar, SegmentedBarItem, ShorthandProperty, Slider, Span, StackLayout, Style, SwipeDirection, Switch, TabView, TabViewItem, TextBase, TextField, TextView, TimePicker, TouchAction, Trace, Transition, Utils, View, ViewBase, ViewHelper, VirtualArray, WebView, WrapLayout, WrappedValue, XmlParser, _setResolver, _updateCharactersInRangeReplacementString, action, addTaggedAdditionalCSS, addWeakEventListener, alert, androidDynamicElevationOffsetProperty, androidElevationProperty, animationTimingFunctionConverter, autocapitalizationTypeProperty, autocorrectProperty, backgroundColorProperty, backgroundImageProperty, backgroundInternalProperty, backgroundPositionProperty, backgroundRepeatProperty, backgroundSizeProperty, booleanConverter, borderBottomColorProperty, borderBottomLeftRadiusProperty, borderBottomRightRadiusProperty, borderBottomWidthProperty, borderLeftColorProperty, borderLeftWidthProperty, borderRightColorProperty, borderRightWidthProperty, borderTopColorProperty, borderTopLeftRadiusProperty, borderTopRightRadiusProperty, borderTopWidthProperty, capitalizationType, clipPathProperty, colorProperty, confirm, eachDescendant, editableProperty, encoding, fontFamilyProperty, fontInternalProperty, fontScaleProperty, fontSizeProperty, fontStyleProperty, fontWeightProperty, fromObject, fromObjectRecursive, getAncestor, getCurrentPage, getFileAccess, getRootLayout, getTransformedText, getViewById, heightProperty, hintProperty, horizontalAlignmentProperty, iOSApplication, inputType, isAndroid, isIOS, isUserInteractionEnabledProperty, keyboardTypeProperty, knownFolders, letterSpacingProperty, lineHeightProperty, login, makeParser, makeValidator, marginBottomProperty, marginLeftProperty, marginRightProperty, marginTopProperty, maxLengthProperty, minHeightProperty, minWidthProperty, opacityProperty, paddingBottomProperty, paddingLeftProperty, paddingRightProperty, paddingTopProperty, parseCSSShadow, parseKeyframeDeclarations, path, perspectiveProperty, placeholderColorProperty, platformNames, profile, profilingDisable, profilingDumpProfiles, profilingEnable, profilingIsRunning, profilingResetProfiles, profilingStart, profilingStartCPU, profilingStop, profilingStopCPU, profilingTime, profilingUptime, prompt, removeTaggedAdditionalCSS, removeWeakEventListener, resolveFileNameFromUrl, returnKeyTypeProperty, rotateProperty, rotateXProperty, rotateYProperty, sanitizeModuleName, scaleXProperty, scaleYProperty, setActivityCallbacks, textAlignmentProperty, textDecorationProperty, textShadowProperty, textTransformProperty, timeConverter, transformConverter, translateXProperty, translateYProperty, unsetValue, updateTextTriggerProperty, verticalAlignmentProperty, visibilityProperty, whiteSpaceProperty, widthProperty, zIndexProperty, zeroLength)
 @ ./node_modules/@nativescript/firebase/firebase-common.js 2:0-39 33:4-9
 @ ./node_modules/@nativescript/firebase/messaging/messaging.ios.js 3:0-46 154:16-36
 @ ./node_modules/@nativescript/firebase/messaging/index.js 1:0-38 2:25-31
 @ ./app/app.ts 4:0-61 34:4-52
```

```
WARNING in ./node_modules/@nativescript/geolocation/index.ios.js 144:56-74
export 'Enums' (imported as 'Enums') was not found in '@nativescript/core' (possible exports: AbsoluteLayout, AccessibilityEvents, AccessibilityLiveRegion, AccessibilityRole, AccessibilityState, AccessibilityTrait, ActionBar, ActionItem, ActionItems, ActivityIndicator, AndroidApplication, Animation, AnimationCurve, Application, ApplicationSettings, Background, Binding, Builder, Button, CSSHelper, CSSType, CSSUtils, ChangeType, CoercibleProperty, Color, Connectivity, ContainerView, ContentView, ControlStateChangeListener, CoreTypes, CssAnimationParser, CssAnimationProperty, CssProperty, CustomLayoutView, DatePicker, Device, DialogStrings, Dialogs, DockLayout, EditableTextBase, File, FileSystemEntity, FlexboxLayout, Folder, Font, FontScaleCategory, FormattedString, Frame, GestureStateTypes, GestureTypes, GesturesObserver, GridLayout, GridUnitType, HtmlView, Http, IOSHelper, Image, ImageAsset, ImageCache, ImageSource, InheritedCssProperty, InheritedProperty, ItemSpec, KeyframeAnimation, KeyframeAnimationInfo, KeyframeDeclaration, KeyframeInfo, Label, LayoutBase, Length, ListPicker, ListView, ModuleNameResolver, NavigationButton, Observable, ObservableArray, Page, PageBase, ParserEvent, ParserEventType, PercentLength, Placeholder, Progress, Property, ProxyViewContainer, PseudoClassHandler, Repeater, RootLayout, Screen, ScrollView, SearchBar, SegmentedBar, SegmentedBarItem, ShorthandProperty, Slider, Span, StackLayout, Style, SwipeDirection, Switch, TabView, TabViewItem, TextBase, TextField, TextView, TimePicker, TouchAction, Trace, Transition, Utils, View, ViewBase, ViewHelper, VirtualArray, WebView, WrapLayout, WrappedValue, XmlParser, _setResolver, _updateCharactersInRangeReplacementString, action, addTaggedAdditionalCSS, addWeakEventListener, alert, androidDynamicElevationOffsetProperty, androidElevationProperty, animationTimingFunctionConverter, autocapitalizationTypeProperty, autocorrectProperty, backgroundColorProperty, backgroundImageProperty, backgroundInternalProperty, backgroundPositionProperty, backgroundRepeatProperty, backgroundSizeProperty, booleanConverter, borderBottomColorProperty, borderBottomLeftRadiusProperty, borderBottomRightRadiusProperty, borderBottomWidthProperty, borderLeftColorProperty, borderLeftWidthProperty, borderRightColorProperty, borderRightWidthProperty, borderTopColorProperty, borderTopLeftRadiusProperty, borderTopRightRadiusProperty, borderTopWidthProperty, capitalizationType, clipPathProperty, colorProperty, confirm, eachDescendant, editableProperty, encoding, fontFamilyProperty, fontInternalProperty, fontScaleProperty, fontSizeProperty, fontStyleProperty, fontWeightProperty, fromObject, fromObjectRecursive, getAncestor, getCurrentPage, getFileAccess, getRootLayout, getTransformedText, getViewById, heightProperty, hintProperty, horizontalAlignmentProperty, iOSApplication, inputType, isAndroid, isIOS, isUserInteractionEnabledProperty, keyboardTypeProperty, knownFolders, letterSpacingProperty, lineHeightProperty, login, makeParser, makeValidator, marginBottomProperty, marginLeftProperty, marginRightProperty, marginTopProperty, maxLengthProperty, minHeightProperty, minWidthProperty, opacityProperty, paddingBottomProperty, paddingLeftProperty, paddingRightProperty, paddingTopProperty, parseCSSShadow, parseKeyframeDeclarations, path, perspectiveProperty, placeholderColorProperty, platformNames, profile, profilingDisable, profilingDumpProfiles, profilingEnable, profilingIsRunning, profilingResetProfiles, profilingStart, profilingStartCPU, profilingStop, profilingStopCPU, profilingTime, profilingUptime, prompt, removeTaggedAdditionalCSS, removeWeakEventListener, resolveFileNameFromUrl, returnKeyTypeProperty, rotateProperty, rotateXProperty, rotateYProperty, sanitizeModuleName, scaleXProperty, scaleYProperty, setActivityCallbacks, textAlignmentProperty, textDecorationProperty, textShadowProperty, textTransformProperty, timeConverter, transformConverter, translateXProperty, translateYProperty, unsetValue, updateTextTriggerProperty, verticalAlignmentProperty, visibilityProperty, whiteSpaceProperty, widthProperty, zIndexProperty, zeroLength)
 @ ./app/shared/utils.ts 2:0-57 307:25-46 316:30-55 317:12-34 331:16-38
 @ ./app/ sync .(xml|js|ts|s?css)$ ./shared/utils.ts
 @ ./app/__@nativescript_webpack_virtual_entry_typescript__ 3:16-92
```

```
WARNING in ./node_modules/@nativescript/geolocation/index.ios.js 301:76-95
export 'Enums' (imported as 'Enums') was not found in '@nativescript/core' (possible exports: AbsoluteLayout, AccessibilityEvents, AccessibilityLiveRegion, AccessibilityRole, AccessibilityState, AccessibilityTrait, ActionBar, ActionItem, ActionItems, ActivityIndicator, AndroidApplication, Animation, AnimationCurve, Application, ApplicationSettings, Background, Binding, Builder, Button, CSSHelper, CSSType, CSSUtils, ChangeType, CoercibleProperty, Color, Connectivity, ContainerView, ContentView, ControlStateChangeListener, CoreTypes, CssAnimationParser, CssAnimationProperty, CssProperty, CustomLayoutView, DatePicker, Device, DialogStrings, Dialogs, DockLayout, EditableTextBase, File, FileSystemEntity, FlexboxLayout, Folder, Font, FontScaleCategory, FormattedString, Frame, GestureStateTypes, GestureTypes, GesturesObserver, GridLayout, GridUnitType, HtmlView, Http, IOSHelper, Image, ImageAsset, ImageCache, ImageSource, InheritedCssProperty, InheritedProperty, ItemSpec, KeyframeAnimation, KeyframeAnimationInfo, KeyframeDeclaration, KeyframeInfo, Label, LayoutBase, Length, ListPicker, ListView, ModuleNameResolver, NavigationButton, Observable, ObservableArray, Page, PageBase, ParserEvent, ParserEventType, PercentLength, Placeholder, Progress, Property, ProxyViewContainer, PseudoClassHandler, Repeater, RootLayout, Screen, ScrollView, SearchBar, SegmentedBar, SegmentedBarItem, ShorthandProperty, Slider, Span, StackLayout, Style, SwipeDirection, Switch, TabView, TabViewItem, TextBase, TextField, TextView, TimePicker, TouchAction, Trace, Transition, Utils, View, ViewBase, ViewHelper, VirtualArray, WebView, WrapLayout, WrappedValue, XmlParser, _setResolver, _updateCharactersInRangeReplacementString, action, addTaggedAdditionalCSS, addWeakEventListener, alert, androidDynamicElevationOffsetProperty, androidElevationProperty, animationTimingFunctionConverter, autocapitalizationTypeProperty, autocorrectProperty, backgroundColorProperty, backgroundImageProperty, backgroundInternalProperty, backgroundPositionProperty, backgroundRepeatProperty, backgroundSizeProperty, booleanConverter, borderBottomColorProperty, borderBottomLeftRadiusProperty, borderBottomRightRadiusProperty, borderBottomWidthProperty, borderLeftColorProperty, borderLeftWidthProperty, borderRightColorProperty, borderRightWidthProperty, borderTopColorProperty, borderTopLeftRadiusProperty, borderTopRightRadiusProperty, borderTopWidthProperty, capitalizationType, clipPathProperty, colorProperty, confirm, eachDescendant, editableProperty, encoding, fontFamilyProperty, fontInternalProperty, fontScaleProperty, fontSizeProperty, fontStyleProperty, fontWeightProperty, fromObject, fromObjectRecursive, getAncestor, getCurrentPage, getFileAccess, getRootLayout, getTransformedText, getViewById, heightProperty, hintProperty, horizontalAlignmentProperty, iOSApplication, inputType, isAndroid, isIOS, isUserInteractionEnabledProperty, keyboardTypeProperty, knownFolders, letterSpacingProperty, lineHeightProperty, login, makeParser, makeValidator, marginBottomProperty, marginLeftProperty, marginRightProperty, marginTopProperty, maxLengthProperty, minHeightProperty, minWidthProperty, opacityProperty, paddingBottomProperty, paddingLeftProperty, paddingRightProperty, paddingTopProperty, parseCSSShadow, parseKeyframeDeclarations, path, perspectiveProperty, placeholderColorProperty, platformNames, profile, profilingDisable, profilingDumpProfiles, profilingEnable, profilingIsRunning, profilingResetProfiles, profilingStart, profilingStartCPU, profilingStop, profilingStopCPU, profilingTime, profilingUptime, prompt, removeTaggedAdditionalCSS, removeWeakEventListener, resolveFileNameFromUrl, returnKeyTypeProperty, rotateProperty, rotateXProperty, rotateYProperty, sanitizeModuleName, scaleXProperty, scaleYProperty, setActivityCallbacks, textAlignmentProperty, textDecorationProperty, textShadowProperty, textTransformProperty, timeConverter, transformConverter, translateXProperty, translateYProperty, unsetValue, updateTextTriggerProperty, verticalAlignmentProperty, visibilityProperty, whiteSpaceProperty, widthProperty, zIndexProperty, zeroLength)
 @ ./app/shared/utils.ts 2:0-57 307:25-46 316:30-55 317:12-34 331:16-38
 @ ./app/ sync .(xml|js|ts|s?css)$ ./shared/utils.ts
 @ ./app/__@nativescript_webpack_virtual_entry_typescript__ 3:16-92
```
**To Reproduce**
Try to import `Enums` from `@nativescript/core` and use one of its properties. 

**Expected behavior**
Enums should still be able to be used as it was in NS7, until it is removed in NS9. 

**Sample project**

**Additional context**

**Comment by basil-wolf** [comment]
Completely agree with the issue and I hope it will be still available for NS9+

**Comment by EddyVerbruggen** [comment]
For anyone ending up here: https://github.com/manijak/nativescript-carousel (iOS) is affected as well.

**Comment by basil-wolf** [comment]
Hi @EddyVerbruggen the firebase plug-in is also giving errors about Enum

**Comment by basil-wolf** [comment]
Thank you @EddyVerbruggen  I hope it can be a quick fix as a lot of developers now looking to move to NS8 as it one of the greatest releases we had in years.

**Comment by NathanWalker** [comment]
This is high priority and expect to have 8.0.1 patch published by Thursday afternoon pst with this adjustment and few others.

**Comment by NathanWalker** [comment]
Fixed in 8.0.1 now https://github.com/NativeScript/NativeScript/releases/tag/8.0.1-core

**Comment by NathanWalker** [comment]
I also posted a PR for @manijak as a convenience to him (it's not needed but provided a note out of consideration):
https://github.com/manijak/nativescript-carousel/pull/189
